### PR TITLE
Update for PureScript 0.11

### PR DESCRIPTION
**Pull request by garyb**

### CODE REVIEW

**Comment by sudhirvkumar** [comment]
@garyb this PR can also be merged?

**Comment by garyb** [comment]
Not quite, there's some changes to come yet.

**Comment by garyb** [comment]
@paf31 Ok, just pushed a pretty big update... 

- Dropped `IsForeign` as discussed variously
- Removed the newtypes for `null` / `undefined` / etc in favour of just providing functions that can be specific about what they read instead - the newtypes only existed to guide the chosen `IsForeign` implementation before.
- Removed `parseJSON` since that's highly misleading about what this library is for (it exists in support code for the examples though, as it makes them easier to read rather than having the example values being parsed in a separate JS file in each case)
- Added a new `!!` operator that allows chaining of property reads, so you can do `value ! prop1 !! prop2`
- Updated all of the examples accordingly

File: src/Data/Foreign/Index.purs [hunk 1]
```diff
@@ -27,19 +28,24 @@ class Index i where
 
 infixl 9 ix as !
 
+ixKleisli :: forall i. Index i => F Foreign -> i -> F Foreign
+ixKleisli f i = f >>= (_ ! i)
+
+infixl 9 ixKleisli as !!
```
Comment (paf31) [inline src/Data/Foreign/Index.purs]: One thought: we could overload `!` using a type class here.

**Pull request merged**